provides.py

```diff
@@ -34,11 +37,34 @@ def set_config(self, is_master, kubeconfig_path):
 
     def config_available(self):
         ''' Ensures all config from the CNI plugin is available. '''
-        cidr = self.all_joined_units.received_raw['cidr']
-        if not cidr:
+        goal_state = hookenv.goal_state()
+        related_apps = [
+            name for name in goal_state['relations'][self.endpoint_name]
+            if '/' not in name
+        ]
+        if not related_apps:
             return False
-        return True
+        configs = self.get_configs()
+        return all(
+            'cidr' in config and 'cni-conf-file' in config
+            for config in [
+                configs.get(related_app, {}) for related_app in related_apps
+            ]
+        )
+
+    def get_config(self, default=None):
+        configs = self.get_configs()
+        if default and default not in configs:
+            msg = 'relation not found for default CNI %s, ignoring' % default
+            hookenv.log(msg, level='WARN')
+            return self.get_config()
+        elif default:
+            return configs.get(default)
+        else:
+            return configs[sorted(configs)[0]]
 
-    def get_config(self):
-        ''' Returns all config from the CNI plugin. '''
-        return self.all_joined_units.received_raw
+    def get_configs(self):
```

Comment:
docstring?  cory was very confused what this does.

Reply:
added